src/commands/sites/sites-create-template.js

```diff
@@ -0,0 +1,251 @@
+// @ts-check
+
+const slugify = require('@sindresorhus/slugify')
+const inquirer = require('inquirer')
+const pick = require('lodash/pick')
+const sample = require('lodash/sample')
+const fetch = require('node-fetch')
+const prettyjson = require('prettyjson')
+const { v4: uuidv4 } = require('uuid')
+
+const { chalk, error, log, logJson, track, warn } = require('../../utils')
+const { getGitHubToken } = require('../../utils/gh-auth')
+
+const SITE_NAME_SUGGESTION_SUFFIX_LENGTH = 5
+
+let ghToken
+
+// Templates are hardcoded for now
+const templates = [
+  {
+    name: 'Next.js Blog theme',
+    sourceCodeUrl: 'https://github.com/netlify-templates/nextjs-blog-theme',
+    slug: 'netlify-templates/nextjs-blog-theme',
+  },
+  {
+    name: 'Next.js Starter',
+    sourceCodeUrl: 'https://github.com/netlify-templates/next-netlify-starter',
+    slug: 'netlify-templates/next-netlify-starter',
+  },
+  {
+    name: 'Nuxt.js Starter',
+    sourceCodeUrl: 'https://github.com/Gomah/bluise',
+    slug: 'Gomah/bluise',
+  },
+  {
+    name: 'Hugo Blog',
+    sourceCodeUrl: 'https://github.com/netlify-templates/one-click-hugo-cms',
+    slug: 'netlify-templates/one-click-hugo-cms',
+  },
+]
+
+/**
+ * The sites:create-template command
+ * @param {import('commander').OptionValues} options
+ * @param {import('../base-command').BaseCommand} command
+ */
+const sitesCreate = async (options, command) => {
+  const { api } = command.netlify
+
+  await command.authenticate()
+
+  console.log(
+    `Choose one of our starter templates. Netlify will create a new repo for this template in your GitHub account.`,
+  )
+
+  const templateUrl = await inquirer.prompt([
+    {
+      type: 'list',
+      name: 'templateName',
+      message: 'Template:',
+      choices: templates.map((template) => ({
+        value: template.slug,
+        name: template.name,
+      })),
+    },
+  ])
+
+  const accounts = await api.listAccountsForUser()
+
+  let { accountSlug } = options
+  if (!accountSlug) {
+    const { accountSlug: accountSlugInput } = await inquirer.prompt([
+      {
+        type: 'list',
+        name: 'accountSlug',
+        message: 'Team:',
+        choices: accounts.map((account) => ({
+          value: account.slug,
+          name: account.name,
+        })),
+      },
+    ])
+    accountSlug = accountSlugInput
+  }
+
+  const { name: nameFlag } = options
+  let user
+  let site
+
+  // Allow the user to reenter site name if selected one isn't available
+  const inputSiteName = async (name) => {
```

Comment:
How is this function different from https://github.com/netlify/cli/blob/40b1cf9b45e78d07817829323320ea8a90c476e4/src/commands/sites/sites-create.js#L49?

Can we re-use the existing code?

Reply:
The first part of this function is identical so I extracted it.